rosbridge_library/src/rosbridge_library/capabilities/defragmentation.py

```diff
@@ -169,7 +169,12 @@ def defragment(self, message):
             self.protocol.log("debug", log_msg)
 
             # Reconstruct the message
-            reconstructed_msg = ''.join(self.received_fragments[msg_id]["fragment_list"][0:message["total"]])
+            print(type(self.received_fragments[msg_id]["fragment_list"]))
+            #reconstructed_msg = ''.join(self.received_fragments[msg_id]["fragment_list"][0:message["total"]])
+            reconstructed_msg = ''
+            for key,value in self.received_fragments[msg_id]["fragment_list"].items():
```

Comment:
1. You're not checking the size bound from the message (`msg['total']`) any more
2. Please change your code back to use `''.join()`, as it's significantly faster (see https://waymoot.org/home/python_string/ )